### PR TITLE
Handle worker error if in placeholder mode (i.e. workerruntime not set)

### PR DESCRIPTION
**Pull request by mhoeger**
Resolves https://github.com/Azure/azure-functions-host/issues/5207

### CODE REVIEW

File: test/WebJobs.Script.Tests/Workers/Rpc/RpcFunctionInvocationDispatcherTests.cs [hunk 1]
```diff
@@ -318,13 +318,28 @@ public async Task FunctionDispatcher_DoNot_Restart_ErroredChannels_If_WorkerRunt
...
+        [InlineData(@"node", true, false, false)]
+        [InlineData(@"node", true, true, false)]
+        [InlineData(@"node", false, false, false)]
+        [InlineData(@"java", false, true, false)]
+        [InlineData("", false, true, false)]
+        public async Task FunctionDispatcher_ShouldNotRestartChannel_InPlaceholderMode(string language, bool isWebHostChannel, bool isJobHostChannel, bool expectedResult)
+        {
+            RpcFunctionInvocationDispatcher functionDispatcher = GetTestFunctionDispatcher(isPlaceholderMode: true);
+            await functionDispatcher.InitializeAsync(null);
+            Assert.Equal(expectedResult, functionDispatcher.ShouldRestartWorkerChannel(language, isWebHostChannel, isJobHostChannel));
```
Comment (pragnagopa) [inline test/WebJobs.Script.Tests/Workers/Rpc/RpcFunctionInvocationDispatcherTests.cs]: instead of adding this test, update earlier test to either initilize function dispatcher with a language or null

Comment (mhoeger) [reply test/WebJobs.Script.Tests/Workers/Rpc/RpcFunctionInvocationDispatcherTests.cs]: @pragnagopa - done!

File: test/WebJobs.Script.Tests/Workers/Rpc/RpcFunctionInvocationDispatcherTests.cs [hunk 2]
```diff
@@ -313,16 +313,23 @@ public async Task FunctionDispatcher_DoNot_Restart_ErroredChannels_If_WorkerRunt
...
+        [InlineData(@"node", @"node", true, true, true)]
+        [InlineData(@"node", @"node", false, false, false)]
+        [InlineData(@"node", @"java", false, true, false)]
+        [InlineData(@"node", "", false, true, false)]
+        [InlineData("", "", false, true, false)]
+        [InlineData("", @"node", false, true, false)]
+        [InlineData("", @"node", true, false, false)]
+        [InlineData("", @"node", true, true, false)]
+        [InlineData("", @"node", false, false, false)]
+        [InlineData("", @"java", false, true, false)]
```
Comment (pragnagopa) [inline test/WebJobs.Script.Tests/Workers/Rpc/RpcFunctionInvocationDispatcherTests.cs]: nit: add case for null workerRunime. Also, remove @ for strings :)

Comment (mhoeger) [reply test/WebJobs.Script.Tests/Workers/Rpc/RpcFunctionInvocationDispatcherTests.cs]: @pragnagopa - removed @! also - removing the "" and using null instead, as _workerRuntime will always either be a value or will be null and not be "".
